experimental/packages/opentelemetry-sdk-metrics-base/src/state/MetricCollector.ts

```diff
@@ -32,12 +32,15 @@ export class MetricCollector implements MetricProducer {
     this.aggregatorTemporality = this._metricReader.getPreferredAggregationTemporality();
   }
 
-  async collect(): Promise<MetricData[]> {
+  async collect(): Promise<ResourceMetrics> {
     const collectionTime = hrTime();
-    const results = await Promise.all(this._sharedState.meters
-      .map(meter => meter.collect(this, collectionTime)));
+    const instrumentationLibraryMetrics = (await Promise.all(this._sharedState.meters
+      .map(meter => meter.collect(this, collectionTime)))).filter(({ metrics }) => metrics.length > 0);
```

Comment:
I'd find we don't need to do this filtering in the SDK. Like, say, with the DELTA aggregation temporality, the metrics for a meter may contain no items if in a period of time there is no metric event reported. It may be confusing that the InstrumentationLibrary sometimes disappears.

Reply:
So you are suggesting to export empty InstrumentationLibraryMetrics in that case and leave it up to the exporter and/or backend to sort it out?

Reply:
Yes, exactly.

Reply:
Removed the filtering